pkg/lib/index/indexer.go

```diff
@@ -0,0 +1,129 @@
+package index
+
+import (
+	"os"
+
+	"github.com/operator-framework/operator-registry/pkg/containertools"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	defaultDockerfileName = "index.Dockerfile"
+	defaultImageTag = "operator-registry-index:latest"
+)
+
+// Indexer allows the creation of index container images from scratch or
+// based on previous index images
+type Indexer interface {
```

Comment:
This feels like it should be two interfaces, one for adding and one for removing?

Reply:
good point